languages/js/src/Host.ts

```diff
@@ -89,6 +89,52 @@ export class Host {
     return cls;
   }
 
+  /**
+   * Get user type for `cls`.
+   */
+  getType(cls: Class):
+    | {
+        name: string;
+        id: number;
+        cls: Class;
+      }
+    | undefined {
+    // TODO: Update when adding user type.
```

Comment:
might be covered by the TODO comment, but I'd prefer something like

```suggestion
  getType(cls: Class): UserType? {
```

Reply:
Even if we aren't introducing `UserType` yet, it would be nice to package `{ name: string; id: number; cls: Class; }` in a `type` that we can use in the return type for `getType` and `distinctUserTypes`